tests/cross/go-build/task.yaml

```
@@ -34,12 +34,17 @@ prepare: |
     cp -ar "$PROJECT_PATH" /tmp/cross-build/src/github.com/snapcore
     chown -R test:12345 /tmp/cross-build
 
+    UBUNTU_ARCHIVE="http://gce.clouds.archive.ubuntu.com/ubuntu/"
+    if [ "$SPREAD_BACKEND" = "google" ]; then
+        UBUNTU_ARCHIVE="http://gce.clouds.archive.ubuntu.com/ubuntu/"
```

no, AFAICT that's wrong; that archive (if it is an archive) is quite a lot further away than the one I think we want.
Try this:
```suggestion
        UBUNTU_ARCHIVE="http://$(cloud-id -l | cut -f2).gce.archive.ubuntu.com/ubuntu/"
```
the one you put is about as far away as archive.u.c (~30ms), whereas this method gets you something that's ~.5ms away.

---

@chipaca nice, change done, thanks for the comment!!